### PR TITLE
update to variable parent messages

### PR DESCRIPTION
**Pull request by Thoralf-M**

### CODE REVIEW

**Comment by Thoralf-M** [comment]
Do we need to update the MilestoneEssence?

File: bindings/nodejs/native/src/classes/client/api.rs [hunk 1]
```diff
@@ -172,31 +169,21 @@ impl Task for ClientTask {
...
-                        let parent1 = match &message.parent1 {
-                            Some(id) => MessageId::from_str(&id)?,
-                            None => tips.0,
-                        };
-                        let parent2 = match &message.parent2 {
-                            Some(id) => MessageId::from_str(&id)?,
-                            None => tips.1,
-                        };
-                        (parent1, parent2)
+                    let parent_msg_ids = if message.parents.is_none() {
```
Comment (lucasfernog) [inline bindings/nodejs/native/src/classes/client/api.rs]: can you change this to use a `match` instead?

File: bindings/nodejs/tests/assertions.js [hunk 2]
```diff
@@ -1,13 +1,5 @@
 const assert = require('assert')
 
-function assertMessage(message) {
```
Comment (lucasfernog) [inline bindings/nodejs/tests/assertions.js:3]: why did you remove this?

File: iota-client/Cargo.toml [hunk 3]
```diff
@@ -13,10 +13,13 @@ license = "Apache-2.0"
 name = "iota_client"
 
 [dependencies]
-bee-rest-api = { git = "https://github.com/iotaledger/bee.git", branch = "chrysalis-pt-2" }
+bee-rest-api = { git = "https://github.com/thibault-martinez/bee.git", branch = "8-parents" }
```
Comment (lucasfernog) [inline iota-client/Cargo.toml:16]: remember to get back to iotaledger/bee when the PR is merged

File: iota-core/Cargo.toml [hunk 4]
```diff
@@ -13,9 +13,11 @@ name = "iota"
 
 [dependencies]
 bee-common = { git = "https://github.com/iotaledger/bee.git", branch = "dev" }
-bee-message = { git = "https://github.com/iotaledger/bee.git", branch = "chrysalis-pt-2" }
+# bee-message = { git = "https://github.com/iotaledger/bee.git", branch = "chrysalis-pt-2" }
+bee-message = { git = "https://github.com/thibault-martinez/bee.git", branch = "8-parents" }
```
Comment (lucasfernog) [inline iota-core/Cargo.toml:17]: remember to use iotaledger/bee when it is merged

**Review by lucasfernog — CHANGES_REQUESTED** [review]

File: bindings/nodejs/tests/assertions.js (hunk 2 above)
Comment (Thoralf-M) [reply bindings/nodejs/tests/assertions.js:3]: Wasn't sure how useful it is, will bring it back only with parents

File: bindings/nodejs/tests/client.js [hunk 5]
```diff
@@ -40,26 +40,27 @@ describe('Client', () => {
...
       .seed(seed)
       .accountIndex(0)
       .output(depositAddress, 2)
       .submit()
-    assertMessageId(messageId)
+    assertMessage(message)
 
     while (true) {
-      const metadata = await client.getMessage().metadata(messageId)
+      const metadata = await client.getMessage().metadata(message.id())
```
Comment (Thoralf-M) [inline bindings/nodejs/tests/client.js:63]: Will fix this in another PR

File: bindings/nodejs/tests/assertions.js (hunk 2 above)
Comment (lucasfernog) [reply bindings/nodejs/tests/assertions.js:3]: it's a utility function for the unit tests

**Review by lucasfernog — APPROVED** [review]
